# Class 5 While loop and functions basic

In [1]:
a=1
b= 10

In [6]:
a=1
b= 10
while a <=b:
    print(a)
    a=a+1
    

1
2
3
4
5
6
7
8
9
10


In [7]:
a=1
b=10

while a<=b:
    print(a)
    a=a+2
    
else :
        print(" print the else block")

1
3
5
7
9
 print the else block


In [15]:
a=1
b=10

while a<= b:
    print(a)
    if a==5:
        break
    a=a+1
else: 
    print(" print the else block")  

1
2
3
4
5


In [1]:
a=1
b=10
while a<b:
    a= a+1
    if a==3:  
        continue
    print(a)
   

2
4
5
6
7
8
9
10


In [3]:
l=[1,2,3,4,65,56]

In [5]:
a=0

while a < len(l):
    print(l[a])
    a=a +1

1
2
3
4
65
56


In [10]:
l

[1, 2, 3, 4, 65, 56]

In [50]:
t=(6,4,5,6,7,8,6)

In [5]:
a=0

while a <= len(l):
    if t[a]==6 or t[a]==7:
        print(a)
        
    a=a+1

NameError: name 'l' is not defined

In [6]:
a=len(t)-8
# make this program generalise i.e remove 8 and 0
while a<=0:
    if t[a]== 6 or t[a]==7:
        print("index is",a)
    if a==-len(t):
        break
    a=a-1

index is -1
index is -3
index is -4
index is -7


In [1]:
d={"k1":"shud", "k2":[1,2,3,4],"k3":(2,3,4,5)}

In [4]:
a=0
b= list(d.keys())

while a< len(d):
    print(d[b[a]])
    a=a+1

shud
[1, 2, 3, 4]
(2, 3, 4, 5)


In [19]:
# Q1. Write a while loop that adds all the numbers up to 100 (inclusive).

a=0
b=0

while a<= 100:
    a=a+1
    b=b+a
    print(b )

1
3
6
10
15
21
28
36
45
55
66
78
91
105
120
136
153
171
190
210
231
253
276
300
325
351
378
406
435
465
496
528
561
595
630
666
703
741
780
820
861
903
946
990
1035
1081
1128
1176
1225
1275
1326
1378
1431
1485
1540
1596
1653
1711
1770
1830
1891
1953
2016
2080
2145
2211
2278
2346
2415
2485
2556
2628
2701
2775
2850
2926
3003
3081
3160
3240
3321
3403
3486
3570
3655
3741
3828
3916
4005
4095
4186
4278
4371
4465
4560
4656
4753
4851
4950
5050
5151


In [1]:
lst=[10, 99, 98, 85, 45, 59, 65, 66, 76, 12, 35, 13, 100, 80, 95]

In [2]:
i = 0
while i < len(lst):
    if lst[i] == 100:
        print("There is a 100 at index no: " + str(i))
    i = i+1

There is a 100 at index no: 12


In [4]:
#Q3. Using while loop and an if statement write a function named name_adder which appends all
the elements in a list to a new list unless the element is an empty string: "".

In [23]:
lst1=["Joe", "Sarah", "Mike", "Jess", "", "Matt", "", "Greg"]

In [28]:
lst1=["Joe", "Sarah", "Mike", "Jess", "", "Matt", "", "Greg"]

#Type your code here.

def name_adder(list):
    i = 0
    new_list = []
    while i < len(list):
        if list[i] != "":
            new_list.append(list[i])
        i = i+1
    return new_list



print(name_adder(lst1))


['Joe', 'Sarah', 'Mike', 'Jess', 'Matt', 'Greg']


In [ ]:
#Q.4 This time inside a function named name_adder, write a while loop that stops appending items to the new list as soon as it encounters an empty string: "".
And prints "There is an empty string and returns the new list."

In [11]:
lst1=["Sam", "", "Ben", "Olivia", "Alicia"]

def name_adder(list):
    i=0
    new_list=[]
    
    while i<len(list):
        if list[i]!=" ":
            new_list.append(list[i])
        else:
            break
            
        return new_list 
        i=i+1
print(name_adder(lst1))
    
    

['Sam']


# Functions

In [3]:
def training():
    print("this is my first function")

In [5]:
training()

this is my first function


In [12]:
training + "nilam"

TypeError: unsupported operand type(s) for +: 'function' and 'str'

In [14]:
type(training())

this is my first function


NoneType

In [15]:
type(training)

function

In [16]:
def test1():
    return(" this is my first funtion")

In [18]:
test1()

' this is my first funtion'

In [19]:
type(test1())

str

In [28]:
test1() + " , nilam"

' this is my first funtion , nilam'

In [29]:
def test2():
    pass

In [30]:
test2()

In [52]:
def test3():
    return 1,3,5, [1,2,3,4,5,67]

In [53]:
test3() # function returns multiple objects in the form of tuple

(1, 3, 5, [1, 2, 3, 4, 5, 67])

In [54]:
a= test3()

In [55]:
a

(1, 3, 5, [1, 2, 3, 4, 5, 67])

In [56]:
a,b,c,d=test3()

In [57]:
a

1

In [58]:
b

3

In [59]:
c

5

In [60]:
d

[1, 2, 3, 4, 5, 67]

In [61]:
def test4():
    a=5*4
    b=6+5
    return a,b

In [62]:
m=test4()

In [63]:
m

(20, 11)

In [64]:
j,k=test4()

In [65]:
j

20

In [66]:
k

11

In [67]:
_, l= test4()

In [68]:
_

20

In [69]:
l

11

In [70]:
test3()

(1, 3, 5, [1, 2, 3, 4, 5, 67])

In [71]:
_,_,_,g=test3()

In [72]:
_

5

In [73]:
_

5

In [74]:
g

[1, 2, 3, 4, 5, 67]

In [77]:
def test5(a,b):
    while a<=b:
        print(a)
        a=a+2
    else:
        print("Print this else block")

In [79]:
test5(1,10)

1
3
5
7
9
Print this else block


In [80]:
test5(10,3)

Print this else block


In [83]:
def test6(a,b):
    while a<=b:
        return (a)
        a=a+1
    else:
        print("print this else block")

In [84]:
test6(2,8)

2

In [86]:
test6(8,2)

print this else block


In [1]:
def test7(a,b):
    l=[]
    while a<=b:
        l.append(a)
        a=a+2
        print(l)
    else:
        print("print the else block")

In [5]:
test7(9,11)

[9]
[9, 11]
print the else block


In [6]:
l= [4,5,6,67,7,8,8,9,9,9,9,[3,4,5,6,7,7,],"sudh"]

In [8]:
l1=[]

for i in l:
    if type(i)== int:
        l1.append(i)
       

In [9]:
l1

[4, 5, 6, 67, 7, 8, 8, 9, 9, 9, 9]

In [12]:
def test8(a,b):
    l1=[]
    for i in l:
        if type(i)== int:
            l1.append(i)
            
        return l1

In [14]:
l1

[4, 5, 6, 67, 7, 8, 8, 9, 9, 9, 9]

In [15]:
l3=[345,34,5,6,7,"sudh",(3,4,56,5)]

In [42]:
def test9(l):
    l1 =[]
    for i in l:
        if type (i)==int:
            l1.append(i)
        
    return(l1)

In [43]:
test9(l3)

[345, 34, 5, 6, 7]

In [48]:
def test10(t):
    a=-1
    while a>= -len(t):
        if t[a]==6 or t[a]== 7:
            print("index :" , a)
        
        if a==-len(t):
            break
        a=a-1   
    

In [51]:
test10(t)

index : -1
index : -3
index : -4
index : -7


In [55]:
t11=(4,5,6,7,8,9,1,2,3,4,5,10)

In [56]:
test10(t11)

index : -9
index : -10


In [75]:
l=[1,11,1,1,1,1,1,11,1]

In [78]:
for i in l:
    if i==1:
         l.remove(1)
    print(l)

[11, 1, 11, 1]
[11, 11, 1]
[11, 11]


In [79]:
l

[11, 11]

In [82]:
l=[1,11,1,1,1,1,1,11,1]

for i in l:
    if i==11:
         l.remove(11)
    print(l)

[1, 11, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 11, 1]
[1, 1, 1, 1, 1, 1, 1]


In [83]:
t

(6, 4, 5, 6, 7, 8, 6)

In [85]:
def test11(t):
    a=-1
    while a>= -len(t):
        if t[a]==6 or t[a]==7:
             print("index : " ,a)
        
        if a== -len(t):
            break
        a=a-1
        
    

In [86]:
test11(t)

index :  -1
index :  -3
index :  -4
index :  -7


In [1]:
#Write a program to create a function that takes two arguments, name and age, and print their value.

In [5]:


def demo(name,age):
    
    print(name,age)
    
    

In [8]:
demo("nilam",32)

nilam 32


In [9]:
#Write a program to create function func1() to accept a variable length of arguments and print their value.

In [22]:
def func1(*args):
    for i in args:
        print(i)
        
func1(20, 40, 60)
func1(80, 100)

20
40
60
80
100


In [ ]:
#Write a program to create function calculation() such that it can accept two variables and 
calculate addition and subtraction. Also, it must return both addition and subtraction in a single return call.

In [39]:
def calculations(a,b):
    addition = a+b
    subtraction = a-b
    return(addition,subtraction)
calculations(9,10)

(19, -1)

In [ ]:
#Write a program to create a function show_employee() using the following conditions.

It should accept the employee’s name and salary and display both.
If the salary is missing in the function call then assign default value 9000 to salary

In [44]:
def show_employee(name,salary=9000):
    print("name :" ,  name ,  "salary  :" , salary  )
    
show_employee("nilam",5000)
show_employee("sonal")

name : nilam salary  : 5000
name : sonal salary  : 9000


In [ ]:
#Create an inner function to calculate the addition in the following way
Create an outer function that will accept two parameters, a and b
Create an inner function inside an outer function that will calculate the addition of a and b
At last, an outer function will add 5 into addition and return it

In [67]:
def outer_fun(a,b):
    def addition(a,b):
        return(a+b)
    add= addition(a,b)
    return add+5
result= outer_fun(5,10)
print(result)

20


In [ ]:
#Write a program to create a recursive function to calculate the sum of numbers from 0 to 10.

A recursive function is a function that calls itself again and again.

In [70]:
def addition(num):
    if num:
        return num+ addition(num-1)
    else:
        return 0
res = addition(10)
print(res)

55


In [ ]:
#Below is the function display_student(name, age). Assign a new name show_tudent(name, age) 
to it and call it using the new name.

In [2]:
def display_student(name, age):
    print(name,age)
    


In [3]:
display_student("nilam","32")

nilam 32


In [6]:
show_student = display_student
show_student("nilam" ,26)

nilam 26


In [7]:
 #Generate a Python list of all the even numbers between 4 to 30

In [10]:
list(range(4,30,2))

[4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28]

In [11]:
#Find the largest item from a given list
x = [4, 6, 8, 24, 12, 2]

In [12]:
max(x)

24

In [25]:
def test(a,b):
    print("this is my function for cocatenation or additon")
    return(a+b)

In [26]:
test(9,10)

this is my function for cocatenation or additon


19

In [27]:
test("sudh", "kumar")

this is my function for cocatenation or additon


'sudhkumar'

In [28]:
test(a=9,b=12)

this is my function for cocatenation or additon


21

In [30]:
test("nilam","46")

this is my function for cocatenation or additon


'nilam46'

In [31]:
def test1(a,b,c,d):
    return(a,b,c,d)

In [33]:
test1(4,5,6,7,9,10)

TypeError: test1() takes 4 positional arguments but 6 were given

In [34]:
def test1(*args): # args is just a variable
    return(args)

In [35]:
test1(4,5,6,7,8,9,10)

(4, 5, 6, 7, 8, 9, 10)

In [37]:
l=[1,11,1,1,1,1,1,11,1]

In [47]:
def test1(*a):
    """this function takes n no of arguments and returns a list contaning all arguments"""
    l =[]
    for i in a:
        l.append(i)
    return(l)

In [50]:
test1(4,5,8,9,0)

[4, 5, 8, 9, 0]

In [54]:
def test2(a,b,c,d ,*m):
    return(a,b,c,d,m)

In [55]:
test2(5,6,7,8,3,6,4,9,0)

(5, 6, 7, 8, (3, 6, 4, 9, 0))

In [56]:
def test3(*m,a,b,c,d): #test2 is possible not test3
    return(m,a,b,c,d)

In [57]:
test3(4,5,7,8,9,10,77,88,99)

TypeError: test3() missing 4 required keyword-only arguments: 'a', 'b', 'c', and 'd'

In [58]:
test3(5,8,9,10,77,88,a=0,b=9,c=7,d=3)

((5, 8, 9, 10, 77, 88), 0, 9, 7, 3)

In [59]:
def test4(**sudh):
    return(sudh)

In [61]:
test4(9,0,8,7)

TypeError: test4() takes 0 positional arguments but 4 were given

In [62]:
test4(a=7,b=0,c=7,d=5,m=[8,9,0,7,8])

{'a': 7, 'b': 0, 'c': 7, 'd': 5, 'm': [8, 9, 0, 7, 8]}

In [63]:
def test5(*m,**s):
    return(m,s)

In [64]:
test5(7,8,9,0,a=5,b=9)

((7, 8, 9, 0), {'a': 5, 'b': 9})

In [19]:
def test6(*m):
    n=0
    for i in m:
        if type(i) == int:
            n=n +i
    return n

In [20]:
test6(2,3,4,5,6,7,8,9)

44

In [17]:
def test7(*m):
    n=1
    for i in m:
        if type(i) == int:
            n=n *i
    return n

In [18]:
test7(7,8,9)

504

# # ananomyus function i.e lambda function

In [5]:
n= lambda a,b : a+b

In [6]:
n(3,4)

7

In [9]:
b= lambda *m : m

In [10]:
b(8,9,6,4,5)

(8, 9, 6, 4, 5)

In [16]:
t = (3,4,5,6,7,8,909,90)

l1=[]
for i in t:
        l1.append(i)
print(l1)

[3, 4, 5, 6, 7, 8, 909, 90]


In [20]:
[i for i in t]

[3, 4, 5, 6, 7, 8, 909, 90]

In [23]:
s="sudh"

In [24]:
[i for i in s]

['s', 'u', 'd', 'h']

In [25]:
[i for i in range(1,10)]

[1, 2, 3, 4, 5, 6, 7, 8, 9]

In [26]:
[i*i for i in range(1,11)]

[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

In [28]:
l1= []

for i in range(1,10):
        l1.append(i)
print(l1)


[1, 2, 3, 4, 5, 6, 7, 8, 9]


In [29]:
l1= []

for i in range(1,10):
        l1.append(i*2)
print(l1)


[2, 4, 6, 8, 10, 12, 14, 16, 18]


In [30]:
l1= []

for i in range(1,10):
        l1.append(i*i)
print(l1)


[1, 4, 9, 16, 25, 36, 49, 64, 81]


In [38]:
b=lambda *x :[i for i in x]

In [39]:
b(1,3,6,8,9,8,[7,9,0,8])

[1, 3, 6, 8, 9, 8, [7, 9, 0, 8]]

In [40]:
b=lambda *x :[i*2 for i in x]

In [41]:
b(1,3,6,8,9,8)

[2, 6, 12, 16, 18, 16]

In [42]:
b=lambda *x :[i*i for i in x]

In [43]:
b(1,3,6,8,9,8)

[1, 9, 36, 64, 81, 64]

In [44]:
#The special syntax **kwargs in function definitions in python is used to pass a keyworded, 
#variable-length argument list.The special syntax **kwargs in function definitions in python 
#is used to pass a keyworded, variable-length argument list.

In [68]:
def test14(**kwargs):
    count = 0
    print(kwargs)
    print(kwargs.values())
    for v in kwargs.values():
        if type(v) == str or type(v) == list:
            count +=1
    return count
            

In [73]:
test14(a= 7, b= "sudh", c=345, d= "sudh", l=[1,2,3,4,5,"sudh"])

{'a': 7, 'b': 'sudh', 'c': 345, 'd': 'sudh', 'l': [1, 2, 3, 4, 5, 'sudh']}
dict_values([7, 'sudh', 345, 'sudh', [1, 2, 3, 4, 5, 'sudh']])


3

In [74]:
def test15(**kwargs):
    count=0
    print(kwargs)
    print(kwargs.values())
    
    for v in kwargs.values():
        if type(v)== str or type(v)==list:
            count = count+1
    return count
    
    
    
    

In [75]:
test15(a=7,b="nilam",c=345,d="kale", l=[6,7,8,9,10,"sudh"])

{'a': 7, 'b': 'nilam', 'c': 345, 'd': 'kale', 'l': [6, 7, 8, 9, 10, 'sudh']}
dict_values([7, 'nilam', 345, 'kale', [6, 7, 8, 9, 10, 'sudh']])


3

In [76]:
def test16(**kwargs):
    return(list(kwargs.values()))

In [88]:
test16(a=7,b="nilam",c=345,d="kale", l=[6,7,8,9,10,"sudh"])

[7, 'nilam', 345, 'kale', [6, 7, 8, 9, 10, 'sudh']]

# Introduction : Lambda Function
In non-technical language, lambda is an alternative way of defining function. 
You can define function inline using lambda. 
It means you can apply a function to some data using a single line of python code. 
It is called anonymous function as the function can be defined without its name. 
They are a part of functional programming style which focus on readability of code and avoids changing mutable data.

Syntax of Lambda Function:
Lambda arguments: expression

Lambda function can have more than one argument but expression cannot be more than 1. 
The expression is evaluated and returned. 
Example:
addition = lambda x,y: x + y
addition(2,3) 
returns 5

In the above python code, x,y are the arguments and x + y is the expression that gets evaluated and returned.

Difference between Lambda and Def Function:

By using both lambda and def, you can create your own user-defined function in python.

def square(x):
     return x**2
 square(2) 
 returns 4

square = lambda x:x**2
square(2) returns 4

There are some difference between them as listed below:

lambda is a keyword that returns a function object and does not create a 'name'. Whereas def creates name in the local namespace
lambda functions are good for situations where you want to minimize lines of code as you can create function in one line of python code. It is not possible using def
lambda functions are somewhat less readable for most Python users.
lambda functions can only be used once, unless assigned to a variable name.
Lambda functions are used along with built-in functions like filter(), map(), reduce().

map() function:

map functions executes the function object (i.e. lambda or def) for each element and returns a list of the elements modified by the function object. In the code below, we are multiplying each element by 2.

mylist = [1, 2, 3, 4]
map(lambda x : x*2, mylist)
It returns map object. You cannot see the returned values directly. To view the result, you need to wrap it in list( )
list(map(lambda x : x*2, mylist))
Output : [2, 4, 6, 8]

filter() function:
It returns the items where function is true. If none of the element meets condition, it will return nothing. In the code below, we are checking if value is greater than 2.

list(filter(lambda x : x > 2 , mylist))
Output : [3, 4]

It returns filter object. To see the output values, you need to put filter( ) function within list( )
Let's say you have a dictionary and you want to filter them by values of specific keys.

d = {'a': [1, 2, 1], 'b': [3, 4, 1], 'c': [5, 6, 2]}
We are filtering values equal to 1 in key 'a' and greater than 1 in key 'b'.
list(filter(lambda x: x[0]==1 and x[1]>1, zip(d['a'],d['b'])))

Output
[(1, 3)]
Here x[0] refers to d['a'] and x[1] refers to d['b'].

reduce() function:

Syntax of reduce function is as follows :
reduce(function, list or tuple)

from functools import reduce

reduce(lambda x,y: x+y, [1,2,3,4])
It returns 10.


In [57]:
# MAP FUNCTION:

mylist2= [2,3,4,5,6,7,8,9] 

list(map(lambda x: x**3,mylist2 ))

[8, 27, 64, 125, 216, 343, 512, 729]

In [59]:
mylist3= [4, 9, 16, 25, 36, 49, 64, 81]

list(map(lambda x: x**3 , mylist3))

[64, 729, 4096, 15625, 46656, 117649, 262144, 531441]

In [62]:
# Filter function:

mylist2= [2,3,4,5,6,7,8,9] 

list(filter(lambda x: x>6, mylist2))

[7, 8, 9]

In [63]:
mylist3= [4, 9, 16, 25, 36, 49, 64, 81]

list(filter(lambda x: x>25,mylist3))

[36, 49, 64, 81]

In [85]:
d = {'a': [1, 2, 1], 'b': [3, 4, 1], 'c': [5, 6, 2]}
list(filter(lambda x: x[0]==1 and x[1]>1 ,zip(d["a"],d["b"])))

[(1, 3)]

In [87]:
d = {'a': [6,8,9], 'b': [3, 4, 1], 'c': [5, 6, 2]}

list(filter(lambda x: x[1]>3 and x[1>3],zip(d["a"],d["c"])))

[(6, 5), (8, 6)]

In [74]:
from functools import reduce

mylist2= [2,3,4,5,6,7,8,9]

reduce(lambda x,y: x+y,[2,3,4,5,6,7,8,9] )

44

In [75]:
mylist3= [4, 9, 16, 25, 36, 49, 64, 81]

reduce(lambda x,y:x+y ,[4, 9, 16, 25, 36, 49, 64, 81])

284

In [89]:
#q1: try to print a prime number in between 1 to 1000.

lower=1
upper=1000
print("prime number between", lower , "and", upper ,"are :")

for num in range(lower,upper+1):
    
    if num>1:
        for i in range(2,num):
            if (num%i)== 0 :
                 break
        else:
            print(num)

prime number between 1 and 1000 are :
2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97
101
103
107
109
113
127
131
137
139
149
151
157
163
167
173
179
181
191
193
197
199
211
223
227
229
233
239
241
251
257
263
269
271
277
281
283
293
307
311
313
317
331
337
347
349
353
359
367
373
379
383
389
397
401
409
419
421
431
433
439
443
449
457
461
463
467
479
487
491
499
503
509
521
523
541
547
557
563
569
571
577
587
593
599
601
607
613
617
619
631
641
643
647
653
659
661
673
677
683
691
701
709
719
727
733
739
743
751
757
761
769
773
787
797
809
811
821
823
827
829
839
853
857
859
863
877
881
883
887
907
911
919
929
937
941
947
953
967
971
977
983
991
997


In [90]:
#write a func which is equivalent to print function in python.

def test(a,b):
    return(a+b)
test(8,9)

17

In [ ]:
#write a func which is replica of list append, extend and pop function.

q7: try to write a func which can perform read operation from .txt file

In [95]:
l1=[1,11,1,1,1,1,1,11,1]

In [96]:
def test1(*a):
    l1=[]
    for i in a:
        l1.append(i)
    return l1

In [98]:
test1(l1,6,7,3,6)

[[1, 11, 1, 1, 1, 1, 1, 11, 1], 6, 7, 3, 6]

In [102]:
def test2(*a):
    l1=[]
    for i in a:
        l1.extend(i)
    return l1

In [105]:
test2(l1)

[1, 11, 1, 1, 1, 1, 1, 11, 1]

In [142]:
def test3(*a):   # in pop we call index no
    l1=[6,7,8,9,10,12,11] 
    for i in a:
        l1.pop(6)
    return l1

In [143]:
test3(6) 

[6, 7, 8, 9, 10, 12]

In [ ]:
# write a lambda func which can return a concatenation of all the string that we will pass.

In [156]:
z= lambda nilam,kale : nilam * 2
result=z("nilam", 2)
print(result)  

nilamnilam


In [157]:
z= lambda nilam,kale : nilam + kale
result=z("nilam", "kale")
print(result)  

nilamkale


In [158]:
#write a lambda func which will return list of squares of all the data bw 1-100

In [47]:
b= lambda *x :[x*2 for x in range(1,100)]

In [43]:
b(1,2)

[1,
 4,
 9,
 16,
 25,
 36,
 49,
 64,
 81,
 100,
 121,
 144,
 169,
 196,
 225,
 256,
 289,
 324,
 361,
 400,
 441,
 484,
 529,
 576,
 625,
 676,
 729,
 784,
 841,
 900,
 961,
 1024,
 1089,
 1156,
 1225,
 1296,
 1369,
 1444,
 1521,
 1600,
 1681,
 1764,
 1849,
 1936,
 2025,
 2116,
 2209,
 2304,
 2401,
 2500,
 2601,
 2704,
 2809,
 2916,
 3025,
 3136,
 3249,
 3364,
 3481,
 3600,
 3721,
 3844,
 3969,
 4096,
 4225,
 4356,
 4489,
 4624,
 4761,
 4900,
 5041,
 5184,
 5329,
 5476,
 5625,
 5776,
 5929,
 6084,
 6241,
 6400,
 6561,
 6724,
 6889,
 7056,
 7225,
 7396,
 7569,
 7744,
 7921,
 8100,
 8281,
 8464,
 8649,
 8836,
 9025,
 9216,
 9409,
 9604,
 9801]

In [ ]:
# write a 10 different different lambda func with choice of your task

In [46]:
#lambda function:

# q1. write a lambda func which will return list of cubes of all the data bw 1-21:

b= lambda *x :[x**3 for x in range(1,21)]
b(1,2,3,4,5,6,7,8,9)

[1,
 8,
 27,
 64,
 125,
 216,
 343,
 512,
 729,
 1000,
 1331,
 1728,
 2197,
 2744,
 3375,
 4096,
 4913,
 5832,
 6859,
 8000]

In [89]:
#q.3 

addition =lambda x,y : x+y

addition(8,9)

17

In [91]:
#q 4 . Add 10 to argument a, and return the result:

x = lambda a : a+10

x(5)


15

In [92]:
#q 5. Multiply argument a with argument b and return the result:

x=lambda a,b: a*b

x(9,6)

54

In [93]:
#q.6 Summarize argument a, b, and c and return the result:

x= lambda a,b,c:a+b+c

x(3,4,5)

12

# Why Use Lambda Functions?
The power of lambda is better shown when you use them as an anonymous function inside another function.
Say you have a function definition that takes one argument, and that argument will be multiplied with an unknown number:

In [102]:
def myfunc(n):
    return lambda a: a*n

mydoubler =myfunc(9)

print(mydoubler(11))

99


In [107]:
def myfunc(n):
    return lambda x,y: x +y* n
mydoubler= myfunc(9)
print(mydoubler(7,8))

79


In [109]:
def myfunc(m):
    return lambda x,y: x*y*m
mydoubler=myfunc(9)
print(mydoubler(8,3))

216


1. Write a Python program to create a lambda function that adds 15 to a given number passed in as an argument, also create a lambda function that multiplies argument x with argument y and print the result

In [114]:
a= lambda n :n+15
print(a(3))

18


In [113]:
a=lambda x,y:x*y
print(a(8,9))

72


2. Write a Python program to create a function that takes one argument, and that argument will be multiplied with an unknown given number. Go to the editor
Sample Output:
Double the number of 15 = 30
Triple the number of 15 = 45
Quadruple the number of 15 = 60
Quintuple the number 15 = 75

In [124]:
def my_compute(n):
    return lambda x: x*n
result= my_compute(2)
print("double the number of 15 :" ,result(15))
result= my_compute(3)
print("double the number of 15 :" ,result(15))
result= my_compute(4)
print("double the number of 15 :" ,result(15))
result= my_compute(5)
print("double the number of 15 :" ,result(15))

double the number of 15 : 30
double the number of 15 : 45
double the number of 15 : 60
double the number of 15 : 75


3. Write a Python program to sort a list of tuples using Lambda.
Original list of tuples:
[('English', 88), ('Science', 90), ('Maths', 97), ('Social sciences', 82)]
Sorting the List of Tuples:
[('Social sciences', 82), ('English', 88), ('Science', 90), ('Maths', 97)]

In [147]:
mark_list =[('English', 88), ('Science', 90), ('Maths', 97), ('Social sciences', 82)]
print("the original list of tuples : ")
print(mark_list)
mark_list.sort(key=lambda x: x[1])
print(" \nsorting list of tuples: " )
print(mark_list)


the original list of tuples : 
[('English', 88), ('Science', 90), ('Maths', 97), ('Social sciences', 82)]
 
sorting list of tuples: 
[('Social sciences', 82), ('English', 88), ('Science', 90), ('Maths', 97)]


4. Write a Python program to sort a list of dictionaries using Lambda. Go to the editor
Original list of dictionaries :
[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}]
Sorting the List of dictionaries :
[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}]

In [168]:
models = [{'make':'Nokia', 'model':216, 'color':'Black'}, {'make':'Mi Max', 'model':'2', 'color':'Gold'}, {'make':'Samsung', 'model': 7, 'color':'Blue'}]
print("Original list of dictionaries :")
print(models)
sorted_models = sorted(models, key = lambda x: x['color'])
print("\nSorting the List of dictionaries :")
print(sorted_models)

Original list of dictionaries :
[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}]

Sorting the List of dictionaries :
[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}]


5. Write a Python program to filter a list of integers using Lambda. Go to the editor
Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Even numbers from the said list:
[2, 4, 6, 8, 10]
Odd numbers from the said list:
[1, 3, 5, 7, 9]

In [179]:
Integers_list=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print(" the originall list of integers: ")
print(Integers_list)

Even_num =list(filter(lambda x:x%2==0,[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
print("\nEven_num :",Even_num)

odd_num =list(filter(lambda x:x%2!=0,[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
print("\nodd_num :",odd_num)

 the originall list of integers: 
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

Even_num : [2, 4, 6, 8, 10]

odd_num : [1, 3, 5, 7, 9]


6. Write a Python program to square and cube every number in a given list of integers using Lambda. Go to the editor
Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Square every number of the said list:
[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
Cube every number of the said list:
[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

In [210]:
Integers_list= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] 
print("The original list of integers: ")
print(Integers_list)

y=lambda *x: [x**2 for x in range(1,11)]
print("Square every number of the said list: ",y())
y=lambda *x: [x**3 for x in range(1,11)]
print("cube every number of the said list: ",y())


The original list of integers: 
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Square every number of the said list:  [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
cube every number of the said list:  [1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]


7. Write a Python program to find if a given string starts with a given character using Lambda

In [213]:
x= lambda y: True if y.startswith("n") else False
print(x("nilamkale"))

x= lambda y: True if y.startswith("l") else False
print(x("nilamkale"))

True
False


8. Write a Python program to extract year, month, date and time using Lambda.

In [4]:
import datetime
now= datetime.datetime.now()
print(now)

year= lambda x: x.year
month=lambda x: x.month
day=lambda x: x.day
time= lambda x:x.time()
print(year(now)) 
print(month(now))
print(day(now))
print(time(now))
        

2022-10-18 12:53:48.600922
2022
10
18
12:53:48.600922


9. Write a Python program to check whether a given string is number or not using Lambda.

In [15]:
is_num = lambda x: x.replace(".","",1).isdigit()
print(is_num("1234"))
print(is_num("0.234"))
print(is_num('-12547'))
print(is_num('00'))
print(is_num('A001'))
print(is_num('001'))
print("\nPrint checking numbers:")

is_num1= lambda y:is_num(y[1:])if y[0]=="-" else is_num(y)
print(is_num1('-16.4'))
print(is_num1('-24587.11'))
print(is_num1('ASAS'))

True
True
False
True
False
True

Print checking numbers:
True
True
False


10. Write a Python program to create Fibonacci series upto n using Lambda.

In [70]:
from functools import reduce
 
fib_series = lambda n: reduce(lambda x, _: x+[x[-1]+x[-2]],
                                range(n-2), [0, 1])
 
print("Fibonacci series upto 2:")
print(fib_series(2))
print("\nFibonacci series upto 5:")
print(fib_series(5))
print("\nFibonacci series upto 6:")
print(fib_series(6))
print("\nFibonacci series upto 9:")
print(fib_series(9))

Fibonacci series upto 2:
[0, 1]

Fibonacci series upto 5:
[0, 1, 1, 2, 3]

Fibonacci series upto 6:
[0, 1, 1, 2, 3, 5]

Fibonacci series upto 9:
[0, 1, 1, 2, 3, 5, 8, 13, 21]


#11. Write a Python program to find intersection of two given arrays using Lambda. Go to the editor
Original arrays:
[1, 2, 3, 5, 7, 8, 9, 10]
[1, 2, 4, 8, 9]

In [77]:
array_1= [1, 2, 3, 5, 7, 8, 9, 10] 
array_2= [1, 2, 4, 8, 9]
print("original arrays: ")
print(array_1)
print(array_2)

intersection= list(filter(lambda x: x in array_1,array_2))
print("intersection of arrays: ")
print(intersection)

original arrays: 
[1, 2, 3, 5, 7, 8, 9, 10]
[1, 2, 4, 8, 9]
intersection of arrays: 
[1, 2, 8, 9]


12. Write a Python program to rearrange positive and negative numbers in a given array using Lambda. Go to the editor
Original arrays:
[-1, 2, -3, 5, 7, 8, 9, -10]

In [85]:


original_arrays= [-1, 2, -3, 5, 7, 8, 9, -10]
print(" the original arrays: ")
print(original_arrays)

pos_neg= sorted(original_arrays,key = lambda x: 0 if x==0 else -1/x)
print(pos_neg)

 the original arrays: 
[-1, 2, -3, 5, 7, 8, 9, -10]
[2, 5, 7, 8, 9, -10, -3, -1]


13. Write a Python program to count the even, odd numbers in a given array of integers using Lambda. Go to the editor
Original arrays:
[1, 2, 3, 5, 7, 8, 9, 10]

In [103]:
array_nums = [1, 2, 3, 5, 7, 8, 9, 10]
print("Original arrays:")
print(array_nums)
odd_ctr = len(list(filter(lambda x: (x%2 != 0) , array_nums)))
even_ctr = len(list(filter(lambda x: (x%2 == 0) , array_nums)))
print("\nNumber of even numbers in the above array: ", even_ctr)
print("\nNumber of odd numbers in the above array: ", odd_ctr)

Original arrays:
[1, 2, 3, 5, 7, 8, 9, 10]

Number of even numbers in the above array:  3

Number of odd numbers in the above array:  5


14. Write a Python program to find the values of length six in a given list using Lambda. 

In [106]:
weekdays=["monday","tuesday","wednesday","thursday","friday","saturday","sunday"]

days= filter(lambda day: day if len(day)==6 else "",weekdays)
for d in days:
    print(d)

monday
friday
sunday


15. Write a Python program to add two given lists using map and lambda. Go to the editor
Original list:
[1, 2, 3]
[4, 5, 6]

In [108]:
array_1= [1, 2, 3]
array_2= [4, 5, 6]
print("original arrays: ")
print(array_1)
print(array_2)

add_two= list(map(lambda x,y:x+y, array_1,array_2))
print(" two number after adding : " , add_two)


original arrays: 
[1, 2, 3]
[4, 5, 6]
 two number after adding :  [5, 7, 9]


16. Write a Python program to find the second lowest grade of any student(s) from the given names and grades of each student using lists and lambda. Input number of students, names and grades of each student.
Input number of students: 5
Name: S ROY
Grade: 1
Name: B BOSE
Grade: 3
Name: N KAR
Grade: 2
Name: C DUTTA
Grade: 1
Name: G GHOSH
Grade: 1

In [4]:
students = []
sec_name = []
second_low =0
n = int(input("Input number of students: "))
for _ in range(n):
    s_name = input("Name: ")
    score = float(input("Total marks: "))
    students.append([s_name,score])
print("\nNames and Marks of all students:")
print(students)
order =sorted(students, key = lambda x: int(x[1]))
for i in range(n):
    if order[i][1]!= order[0][1]:
        second_low = order[i][1]
    break
print("\nSecond lowest Marks: ",second_low)
sec_student_name = [x[0] for x in order if x[1] == second_low]
sec_student_name.sort()
print("\nNames:")
for s_name in sec_student_name:
    print(s_name)

Input number of students: 5
Name: roy
Total marks: 1
Name: ghosh
Total marks: 3
Name: kar
Total marks: 2
Name: dutta
Total marks: 1
Name: bose
Total marks: 1

Names and Marks of all students:
[['roy', 1.0], ['ghosh', 3.0], ['kar', 2.0], ['dutta', 1.0], ['bose', 1.0]]

Second lowest Marks:  0

Names:


17. Write a Python program to find numbers divisible by nineteen or thirteen from a list of numbers using Lambda. Go to the editor
Orginal list:
[19, 65, 57, 39, 152, 639, 121, 44, 90, 190]
Numbers of the above list divisible by nineteen or thirteen:
[19, 65, 57, 39, 152, 190]

In [13]:
Original_list = [19, 65, 57, 39, 152, 639, 121, 44, 90, 190]
print(Original_list)

divisible= list(filter(lambda x: x%19 == 0 or x%13==0,Original_list))

print(divisible)


[19, 65, 57, 39, 152, 639, 121, 44, 90, 190]
[19, 65, 57, 39, 152, 190]


 18. Write a Python program to find palindromes in a given list of strings using Lambda. Orginal list of strings:
['php', 'w3r', 'Python', 'abcd', 'Java', 'aaa']

In [30]:
Orginal_list_of_strings =['php', 'w3r', 'Python', 'abcd', 'Java', 'aaa']
print(Orginal_list_of_strings)
result = list(filter(lambda x: x== "".join(reversed(x)),Orginal_list_of_strings))
print(result)    
    
    

['php', 'w3r', 'Python', 'abcd', 'Java', 'aaa']
['php', 'aaa']


19. Write a Python program to find all anagrams of a string in a given list of strings using lambda. Go to the editor
Orginal list of strings:
['bcda', 'abce', 'cbda', 'cbea', 'adcb']

In [10]:

from collections import Counter
Orginal_list = ['bcda', 'abce', 'cbda', 'cbea', 'adcb']
str=("bcda")
print(Orginal_list)

results= list(filter(lambda x: (Counter(str)== Counter(x)),Orginal_list ))
print( "\nthe list of Anagrams: ")
print(results)

['bcda', 'abce', 'cbda', 'cbea', 'adcb']

the list of Anagrams: 
['bcda', 'cbda', 'adcb']


20. Write a Python program to find the numbers of a given string and store them in a list, display the numbers which are bigger than the length of the list in sorted form. Use lambda function to solve the problem. Go to the editor
Original string: sdf 23 safs8 5 sdfsd8 sdfs 56 21sfs 20 5

In [13]:
str1 = "sdf 23 safs8 5 sdfsd8 sdfs 56 21sfs 20 5"
print("Original string: ",str1)
str_num=[i for i in str1.split(' ')]
lenght=len(str_num)
numbers=sorted([int(x) for x in str_num if x.isdigit()])
print('Numbers in sorted form:')
for i in ((filter(lambda x:x>lenght,numbers))):
    print(i,end=' ')

Original string:  sdf 23 safs8 5 sdfsd8 sdfs 56 21sfs 20 5
Numbers in sorted form:
20 23 56 

21. Write a Python program that multiply each number of given list with a given number using lambda function. Print the result. Go to the editor
Original list: [2, 4, 6, 9, 11]
Given number: 2
Result:
4 8 12 18 22

In [16]:
Original_list= [2, 4, 6, 9, 11] 
print(Original_list)

result= list(map(lambda x:x*2,Original_list))
print(result)

[2, 4, 6, 9, 11]
[4, 8, 12, 18, 22]


22. Write a Python program that sum the length of the names of a given list of names after removing the names that starts with an lowercase letter. Use lambda function

In [27]:
sample_names = ['sally', 'Dylan', 'rebecca', 'Diana', 'Joanne', 'keith']
print(sample_names)

results= list(filter(lambda x: x[0].isupper() and x[1:].islower(),sample_names ))
print(results)
print( len("".join(results)))

['sally', 'Dylan', 'rebecca', 'Diana', 'Joanne', 'keith']
['Dylan', 'Diana', 'Joanne']
16


23. Write a Python program to calculate the sum of the positive and negative numbers of a given list of numbers using lambda function. Go to the editor
Original list: [2, 4, -6, -9, 11, -12, 14, -5, 17]

In [33]:
Original_list=[2, 4, -6, -9, 11, -12, 14, -5, 17]
print(Original_list)

result_positive= list(filter(lambda x:x>0 ,Original_list))
result_negative= list(filter(lambda x:x<0 ,Original_list))
print(" the sum of positive number: ", sum(result_positive))
print(" the sum of negative number: ", sum(result_negative))

[2, 4, -6, -9, 11, -12, 14, -5, 17]
 the sum of positive number:  48
 the sum of negative number:  -32


24. Write a Python program to find numbers within a given range where every number is divisible by every digit it contains.

In [10]:
def divisible_by_number(start_num,end_num):
    return [n for n in range(start_num,end_num+1) \
            if not any(map(lambda x: int(x)==0 or n%int(x)!=0, str(n)))]


In [11]:
print(divisible_by_number(1,22))

[1, 2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 15, 22]


25. Write a Python program to create the next bigger number by rearranging the digits of a given number. Go to the editor
Original number: 12
Next bigger number: 21
Original number: 10
Next bigger number: False
Original number: 201
Next bigger number: 210
Original number: 102
Next bigger number: 120
Original number: 445
Next bigger number: 454

In [2]:
def rearrange_bigger(n):
    #Break the number into digits and store in a list
    nums = list(str(n))
    for i in range(len(nums)-2,-1,-1):
        if nums[i] < nums[i+1]:
            z = nums[i:]
            y = min(filter(lambda x: x > z[0], z))
            z.remove(y)
            z.sort()
            nums[i:] = [y] + z
            return int("".join(nums))
    return False
  
n=12
print(" the original number: " ,n)
print(" Next bigger number:",rearrange_bigger(n))
          
n=10
print(" the original number: " ,n)
print(" Next bigger number:",rearrange_bigger(n))
                   
n=201
print(" the original number: " ,n)
print(" Next bigger number:",rearrange_bigger(n))
                   
n=102
print(" the original number: " ,n)
print(" Next bigger number:",rearrange_bigger(n))
                   
n=445
print(" the original number: " ,n)
print(" Next bigger number:",rearrange_bigger(n))
                                      
                   

 the original number:  12
 Next bigger number: 21
 the original number:  10
 Next bigger number: False
 the original number:  201
 Next bigger number: 210
 the original number:  102
 Next bigger number: 120
 the original number:  445
 Next bigger number: 454


Write a Python program to find the list with maximum and minimum length using lambda. Go to the editor
Original list:
[[0], [1, 3], [5, 7], [9, 11], [13, 15, 17]]

In [16]:
original_list=  [[0], [1, 3], [5, 7], [9, 11], [13, 15, 17]]
print(original_list)

def max_length_list(original_list):
    max_length= max(len(i) for i in original_list)
    max_list= max(original_list,key = lambda i: len(i))
    return( max_length,max_list)

def min_length_list(original_list):
    min_length= min(len(i) for i in original_list)
    min_list= min(original_list,key= lambda i: len(i))
    return( min_length,min_list)

print("maximum length of list: ")
print(max_length_list(original_list))
print("minimum length of list: ")
print(min_length_list(original_list))

[[0], [1, 3], [5, 7], [9, 11], [13, 15, 17]]
maximum length of list: 
(3, [13, 15, 17])
minimum length of list: 
(1, [0])


27.Write a Python program to sort each sublist of strings in a given list of lists using lambda. Go to the editor
Original list:
[['green', 'orange'], ['black', 'white'], ['white', 'black', 'orange']]

In [5]:
def sort_sublists(input_list):
    result = [sorted(x, key = lambda x:x[0]) for x in input_list] 
    return result
color1 = [["green", "orange"], ["black", "white"], ["white", "black", "orange"]]
print("\nOriginal list:")
print(color1)  
print("\nAfter sorting each sublist of the said list of lists:")
print(sort_sublists(color1))


Original list:
[['green', 'orange'], ['black', 'white'], ['white', 'black', 'orange']]

After sorting each sublist of the said list of lists:
[['green', 'orange'], ['black', 'white'], ['black', 'orange', 'white']]


28. Write a Python program to sort a given list of lists by length and value using lambda. Go to the editor
Original list:
[[2], [0], [1, 3], [0, 7], [9, 11], [13, 15, 17]]

In [11]:
def length_value(Original_list):
    result = [sorted( Original_list,key = lambda x : (len(x),x))]
    return(result)
Original_list=[[2], [0], [1, 3], [0, 7], [9, 11], [13, 15, 17]]
print(Original_list)

print("\nlist after sorting: ")
print(length_value(Original_list))

[[2], [0], [1, 3], [0, 7], [9, 11], [13, 15, 17]]

list after sorting: 
[[[0], [2], [0, 7], [1, 3], [9, 11], [13, 15, 17]]]


29. Write a Python program to find the maximum value in a given heterogeneous list using lambda. Go to the editor
Original list:
['Python', 3, 2, 4, 5, 'version']

In [15]:
def max_value( Original_list):
    result= max(Original_list,key = lambda x : (isinstance(x,int),x))
    return result
Original_list=['Python', 3, 2, 4, 5, 'version']
print( Original_list)

print("\nmaximum value in the list: ")
print(max_value( Original_list))

['Python', 3, 2, 4, 5, 'version']

maximum value in the list: 
5


30. Write a Python program to sort a given matrix in ascending order according to the sum of its rows using lambda. Go to the editor
Original Matrix:
[[1, 2, 3], [2, 4, 5], [1, 1, 1]]

In [18]:
def ascending_order(Original_Matrix):
    result = sorted( Original_Matrix,key = lambda matrix_sum : sum(matrix_sum))
    return result
Original_Matrix =[[1, 2, 3], [2, 4, 5], [1, 1, 1]]
print(Original_Matrix)

print("\nsum of the matrix row: ")
print(ascending_order(Original_Matrix))

[[1, 2, 3], [2, 4, 5], [1, 1, 1]]

sum of the matrix row: 
[[1, 1, 1], [1, 2, 3], [2, 4, 5]]


31. Write a Python program to extract specified size of strings from a give list of string values using lambda. Go to the editor
Original list:
['Python', 'list', 'exercises', 'practice', 'solution']
length of the string to extract:
8

In [19]:
def extract_size(Original_list,l):
    result= list(filter(lambda x: len(x)== l , Original_list ))
    return result
Original_list = ['Python', 'list', 'exercises', 'practice', 'solution']
print(Original_list)
l=8
print("\n list after extracting:")
print(extract_size(Original_list,l))

['Python', 'list', 'exercises', 'practice', 'solution']

 list after extracting:
['practice', 'solution']


32. Write a Python program to count float number in a given mixed list using lambda. Go to the editor
Original list:
[1, 'abcd', 3.12, 1.2, 4, 'xyz', 5, 'pqr', 7, -5, -12.22]

In [24]:
def count_float(Original_list):
    result= list(map(lambda x: isinstance(x,float), Original_list))
    mixed_list = len([e for e in result if e])
    return mixed_list
Original_list= [1, 'abcd', 3.12, 1.2, 4, 'xyz', 5, 'pqr', 7, -5, -12.22]
print(Original_list)

print("\n count of floating point number: ")
print(count_float(Original_list))
    
    

[1, 'abcd', 3.12, 1.2, 4, 'xyz', 5, 'pqr', 7, -5, -12.22]

 count of floating point number: 
3


33. Write a Python program to check whether a given string contains a capital letter, a lower case letter, a number and a minimum length using lambda.

In [2]:
def check_string(str1):
    messg = [
    lambda str1: any(x.isupper() for x in str1) or 'String must have 1 upper case character.',
    lambda str1: any(x.islower() for x in str1) or 'String must have 1 lower case character.',
    lambda str1: any(x.isdigit() for x in str1) or 'String must have 1 number.',
    lambda str1: len(str1) >= 7                 or 'String length should be atleast 8.',]
    result = [x for x in [i(str1) for i in messg] if x != True]
    if not result:
        result.append('Valid string.')
    return result    
s = input("Input the string: ")
print(check_string(s))
         
         
         
         
         

Input the string:  W3resource
['Valid string.']


34. Write a Python program to filter the height and width of students, which are stored in a dictionary using lambda. Go to the editor
Original Dictionary:
{'Cierra Vega': (6.2, 70), 'Alden Cantrell': (5.9, 65), 'Kierra Gentry': (6.0, 68), 'Pierre Cox': (5.8, 66)}

In [14]:
def filter_data (Original_Dictionary):
        result = dict(filter(lambda x: (x[1][0],x[1][1])>(6.0,70),Original_Dictionary.items()))
        return result
    
Original_Dictionary = {'Cierra Vega': (6.2, 70), 'Alden Cantrell': (5.9, 65), 'Kierra Gentry': (6.0, 68), 'Pierre Cox': (5.8, 66)}
print(Original_Dictionary)
    
print(filter_data (Original_Dictionary))


{'Cierra Vega': (6.2, 70), 'Alden Cantrell': (5.9, 65), 'Kierra Gentry': (6.0, 68), 'Pierre Cox': (5.8, 66)}
{'Cierra Vega': (6.2, 70)}


35. Write a Python program to check whether a specified list is sorted or not using lambda. Go to the editor
Original list:
[1, 2, 4, 6, 8, 10, 12, 14, 16, 17]

In [16]:
 def is_sort_list(nums,key=lambda x:x):
        for i ,e in enumerate (nums[1:]):
            if key(e)< key(nums[i]):
                return False
            return True
    
nums1 = [1,2,4,6,8,10,12,14,16,17]
print ("Original list:")
print(nums1)
print("\nIs the said list is sorted!")
print(is_sort_list(nums1)) 
nums2 = [2,3,8,4,7,9,8,2,6,5,1,6,1,2,3,4,6,9,1,2]
print ("\nOriginal list:")
print(nums2)
print("\nIs the said list is sorted!")
print(is_sort_list(nums2))


Original list:
[1, 2, 4, 6, 8, 10, 12, 14, 16, 17]

Is the said list is sorted!
True

Original list:
[2, 3, 8, 4, 7, 9, 8, 2, 6, 5, 1, 6, 1, 2, 3, 4, 6, 9, 1, 2]

Is the said list is sorted!
True


36. Write a Python program to extract the nth element from a given list of tuples using lambda. Go to the editor
Original list:
[('Greyson Fulton', 98, 99), ('Brady Kent', 97, 96), ('Wyatt Knott', 91, 94), ('Beau Turnbull', 94, 98)]
Extract nth element ( n = 0 ) from the said list of tuples:
['Greyson Fulton', 'Brady Kent', 'Wyatt Knott', 'Beau Turnbull']
Extract nth element ( n = 2 ) from the said list of tuples:
[99, 96, 94, 98]

In [35]:
def extract_n_element(test_list,n):
    result= list(map(lambda x: (x[n]) ,test_list))
    return result

test_list = [('Greyson Fulton', 98, 99), ('Brady Kent', 97, 96), ('Wyatt Knott', 91, 94), ('Beau Turnbull', 94, 98)]
print(test_list)

n=0
print("Extract nth element (n=" ,n==0,")from the said list of tuples:")
print("\n nth element :" ,n)
print(extract_n_element(test_list,n))

n=2
print("Extract nth element (n=",n==2,") from the said list of tuples:")
print("\n nth element :" ,n)
print(extract_n_element(test_list,n))

[('Greyson Fulton', 98, 99), ('Brady Kent', 97, 96), ('Wyatt Knott', 91, 94), ('Beau Turnbull', 94, 98)]
Extract nth element (n= True )from the said list of tuples:

 nth element : 0
['Greyson Fulton', 'Brady Kent', 'Wyatt Knott', 'Beau Turnbull']
Extract nth element (n= True ) from the said list of tuples:

 nth element : 2
[99, 96, 94, 98]


37. Write a Python program to sort a list of lists by a given index of the inner list using lambda.

In [6]:
def index_on_inner_list(list_data,index_no):
    result= sorted(list_data,key=lambda x: x[index_no])
    return result
students = [('Greyson Fulton', 98, 99), ('Brady Kent', 97, 96), ('Wyatt Knott', 91, 94), ('Beau Turnbull', 94, 98)] 
print ("Original list:")
print(students)
index_no = 0
print("\nSort the said list of lists by a given index","(index=",index_no,") of the inner list")
print( index_on_inner_list(students,index_no))
index_no = 2
print("\nSort the said list of lists by a given index","(index=",index_no,") of the inner list")
print( index_on_inner_list(students,index_no))

Original list:
[('Greyson Fulton', 98, 99), ('Brady Kent', 97, 96), ('Wyatt Knott', 91, 94), ('Beau Turnbull', 94, 98)]

Sort the said list of lists by a given index (index= 0 ) of the inner list
[('Beau Turnbull', 94, 98), ('Brady Kent', 97, 96), ('Greyson Fulton', 98, 99), ('Wyatt Knott', 91, 94)]

Sort the said list of lists by a given index (index= 2 ) of the inner list
[('Wyatt Knott', 91, 94), ('Brady Kent', 97, 96), ('Beau Turnbull', 94, 98), ('Greyson Fulton', 98, 99)]


38. Write a Python program to remove all elements from a given list present in another list using lambda. 

In [10]:
def remove_list1_list2(list1,list2):
    result= list(filter(lambda x: x not in list2,list1))
    return result
list1= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
list2=[2, 4, 6, 8]
print("original list:", list1)
print("original list:", list2)

print("\nremove all elements from list1 to list2:",)
print(remove_list1_list2(list1,list2))


original list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
original list: [2, 4, 6, 8]

remove all elements from list1 to list2:
[1, 3, 5, 7, 9, 10]


39. Write a Python program to find the elements of a given list of strings that contain specific substring using lambda

In [24]:
def list_of_substrings(str1, sub_string):
    result= list(filter(lambda x: sub_string in x,str1 ))
    return result
Original_list=['red', 'black', 'white', 'green', 'orange']
print(Original_list)
sub_string="ack"
print("\nsub_string to search:", sub_string)
print("Elements of the said list that contain specific substring:",)
print( list_of_substrings(Original_list,sub_string))

sub_string="abc"
print("\nsub_string to search:", sub_string)
print("Elements of the said list that contain specific substring:",)
print( list_of_substrings(Original_list,sub_string))


['red', 'black', 'white', 'green', 'orange']

sub_string to search: ack
Elements of the said list that contain specific substring:
['black']

sub_string to search: abc
Elements of the said list that contain specific substring:
[]


40. Write a Python program to find the nested lists elements, which are present in another list using lambda. Go to the editor

In [5]:
def nested_list(list_1,list_2):
    result= [list(filter(lambda x: x in list_1 , sub_string )) for sub_string in list_2]
    return result
Original_lists_1= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14] 
Original_lists_2=[[12, 18, 23, 25, 45], [7, 11, 19, 24, 28], [1, 5, 8, 18, 15, 16]]
print("The original list1: ", Original_lists_1)
print("The original list2: ", Original_lists_2)

print("\nthe nested lists elements, which are present in another list: ")
print(nested_list(Original_lists_1,Original_lists_2))



The original list1:  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
The original list2:  [[12, 18, 23, 25, 45], [7, 11, 19, 24, 28], [1, 5, 8, 18, 15, 16]]

the nested lists elements, which are present in another list: 
[[12], [7, 11], [1, 5, 8]]


41. Write a Python program to reverse strings in a given list of string values using lambda. 

In [11]:

def reverse_list(list1):
    result= [list(map(lambda x: "".join(reversed(x)),list1))]
    return result
Original_lists=['Red', 'Green', 'Blue', 'White', 'Black']
print(" the original list:",Original_lists)

print("Reverse strings of the said given list:")
print(reverse_list(Original_lists))


 the original list: ['Red', 'Green', 'Blue', 'White', 'Black']
Reverse strings of the said given list:
[['deR', 'neerG', 'eulB', 'etihW', 'kcalB']]


42. Write a Python program to calculate the product of a given list of numbers using lambda. 

In [18]:
import functools

def product_list(list1):
    result= functools.reduce(lambda x,y: x*y,list1,1)
    return result

list1=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print("the original list:", list1)
print("Product of the said list numbers:")
print( product_list(list1))
list2= [2.2, 4.12, 6.6, 8.1, 8.3]
print("the original list:", list2)
print("Product of the said list numbers:")
print( product_list(list2))

the original list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Product of the said list numbers:
3628800
the original list: [2.2, 4.12, 6.6, 8.1, 8.3]
Product of the said list numbers:
4021.8599520000007


43. Write a Python program to multiply all the numbers in a given list using lambda

In [22]:
import functools

def multiply_list(list1):
    result = functools.reduce(lambda x,y: x*y,list1)
    return result
Original_list=[4, 3, 2, 2, -1, 18]
print("the original list:",Original_list)
print("multiply all the numbers of the said list:")
print(multiply_list(Original_list))
Original_list2 =[2, 4, 8, 8, 3, 2, 9]
print("the original list:",Original_list)
print("multiply all the numbers of the said list:")
print(multiply_list(Original_list2))

the original list: [4, 3, 2, 2, -1, 18]
multiply all the numbers of the said list:
-864
the original list: [4, 3, 2, 2, -1, 18]
multiply all the numbers of the said list:
27648


44. Write a Python program to calculate the average value of the numbers in a given tuple of tuples using lambda. 

In [7]:
def average_tuple(nums):
    result = tuple(map(lambda x: sum(x) / float(len(x)), zip(*nums)))
    return result

Original_Tuple=((10, 10, 10), (30, 45, 56), (81, 80, 39), (1, 2, 3))
print("the original list:",Original_Tuple)

print("Average value of the numbers of the said tuple of tuples:")
print( average_tuple(Original_Tuple))

Original_Tuple1=((1, 1, -5), (30, -15, 56), (81, -60, -39), (-10, 2, 3))
print("the original list:",Original_Tuple1)

print("Average value of the numbers of the said tuple of tuples:")
print( average_tuple(Original_Tuple1))


the original list: ((10, 10, 10), (30, 45, 56), (81, 80, 39), (1, 2, 3))
Average value of the numbers of the said tuple of tuples:
(30.5, 34.25, 27.0)
the original list: ((1, 1, -5), (30, -15, 56), (81, -60, -39), (-10, 2, 3))
Average value of the numbers of the said tuple of tuples:
(25.5, -18.0, 3.75)


45. Write a Python program to convert string element to integer inside a given tuple using lambda.

In [12]:
def con_str_ele(test_tuple):
    result = tuple(map(lambda x:(int(x[0]),int(x[2])) , test_tuple))
    return result

Original_tuple_values=(('233', 'ABCD', '33'), ('1416', 'EFGH', '55'), ('2345', 'WERT', '34'))
print("Original_tuple_values:",Original_tuple_values)

print("the new values of tuples:")
print(con_str_ele(Original_tuple_values))

Original_tuple_values: (('233', 'ABCD', '33'), ('1416', 'EFGH', '55'), ('2345', 'WERT', '34'))
the new values of tuples:
((233, 33), (1416, 55), (2345, 34))


46. Write a Python program to find index position and value of the maximum and minimum values in a given list of numbers using lambda. Go to the editor


In [5]:
def position_max_min(nums):
    max_result = max(enumerate(nums), key=(lambda x: x[1]))
    min_result = min(enumerate(nums), key=(lambda x: x[1]))
    return max_result,min_result

nums = [12,33,23,10.11,67,89,45,66.7,23,12,11,10.25,54]
print("Original list:")
print(nums)
result = position_max_min(nums)
print("\nIndex position and value of the maximum value of the said list:")
print(result[0])
print("\nIndex position and value of the minimum value of the said list:")
print(result[1])

Original list:
[12, 33, 23, 10.11, 67, 89, 45, 66.7, 23, 12, 11, 10.25, 54]

Index position and value of the maximum value of the said list:
(5, 89)

Index position and value of the minimum value of the said list:
(3, 10.11)


47. Write a Python program to sort a given mixed list of integers and strings using lambda. Numbers must be sorted before strings.

In [10]:
def sort_mixed_list(list1):
    list1.sort(key = lambda e: (isinstance(e,str),e))
    return list1

Original_list=[19, 'red', 12, 'green', 'blue', 10, 'white', 'green', 1]
print("Original_list: ",Original_list)

print("Numbers must be sorted before strings:")
print(sort_mixed_list(Original_list))


Original_list:  [19, 'red', 12, 'green', 'blue', 10, 'white', 'green', 1]
Numbers must be sorted before strings:
[1, 10, 12, 19, 'blue', 'green', 'green', 'red', 'white']


48. Write a Python program to sort a given list of strings(numbers) numerically using lambda

In [21]:
def sort_list(list1):
    result = sorted(list1,key = lambda x :int(x))
    return result

list1=['4', '12', '45', '7', '0', '100', '200', '-12', '-500']
print("list1:",list1)

print(sort_list(list1))


list1: ['4', '12', '45', '7', '0', '100', '200', '-12', '-500']
['-500', '-12', '0', '4', '7', '12', '45', '100', '200']


49. Write a Python program to count the occurrences of the items in a given list using lambda

In [24]:
def count_occurences(nums):
    result = dict(map(lambda x: (x, list(nums).count(x)),nums))
    return result

nums =[3, 4, 5, 8, 0, 3, 8, 5, 0, 3, 1, 5, 2, 3, 4, 2]
print(nums)
print("Count the occurrences of the items in the said list:")
print(count_occurences(nums))

[3, 4, 5, 8, 0, 3, 8, 5, 0, 3, 1, 5, 2, 3, 4, 2]
Count the occurrences of the items in the said list:
{3: 4, 4: 2, 5: 3, 8: 2, 0: 2, 1: 1, 2: 2}


50. Write a Python program to remove specific words from a given list using lambda. 

In [31]:
def remove_words(list1,remove_words):
    result= list(filter(lambda words:words not in remove_words,list1))
    return result

colours=['orange', 'red', 'green', 'blue', 'white', 'black']
print(colours)
remove_colours= ['orange', 'black']
print(remove_colours)
print(remove_words(colours,remove_colours))



['orange', 'red', 'green', 'blue', 'white', 'black']
['orange', 'black']
['red', 'green', 'blue', 'white']


51. Write a Python program to find the maximum and minimum values in a given list of tuples using lambda function

In [41]:
def max_min_list_tuples(class_students):
    return_max = max(class_students,key=lambda item:item[1])[1]
    return_min = min(class_students,key=lambda item:item[1])[1]
    return return_max, return_min
    
class_students = [('V', 62), ('VI', 68), ('VII', 72), ('VIII', 70), ('IX', 74), ('X', 65)]
print("Original list with tuples:")
print(class_students)
print("\nMaximum and minimum values of the said list of tuples:")
print(max_min_list_tuples(class_students))

Original list with tuples:
[('V', 62), ('VI', 68), ('VII', 72), ('VIII', 70), ('IX', 74), ('X', 65)]

Maximum and minimum values of the said list of tuples:
(74, 62)


52. Write a Python program to remove None value from a given list using lambda function.

In [51]:
def remove_none(list1,remove_word):
    result= list(filter(lambda word: word is not None,list1 ))
    return result

Original_list=[12, 0, None, 23, None, -55, 234, 89, None, 0, 6, -12]
print(Original_list)

remove_word = ["None"]
print(remove_word)
print(remove_none(Original_list, remove_word))

[12, 0, None, 23, None, -55, 234, 89, None, 0, 6, -12]
['None']
[12, 0, 23, -55, 234, 89, 0, 6, -12]
